### ISSUE TITLE
Units can be held in reserve

### ISSUE THREAD

**Issue by yakryder** [stevegrossi/sengoku#8]
![image](https://user-images.githubusercontent.com/14241275/31057794-5f977ade-a6b6-11e7-8781-b0d3ad3b9aa8.png)

Steps to reproduce:
1) Start a new hot seat game
2) On any turn, do not place armies and click 'End turn'.

Expected: Units either must be placed or cannot be held from one round to another.
Actual: Units can optionally not be placed and are then held from one round to another.

**Comment by stevegrossi** [comment]
Interesting point here. With Sengoku, I'm not necessarily trying to match the behavior of Risk (which doesn't allow armies to be held in reserve), but if letting units carry over from one turn to another makes the game less interesting or fun, I'm all for disallowing it. I'm curious why you expected not to be able to let units carry over.

**Comment by yakryder** [comment]
The expectation was mainly based off of the Risk model. Also, I wasn't sure at first of the underlying mechanism to the the game not ending when all players but one no longer have units on the map. My understanding now is that the defeat condition for a player is no owned territories.

**Comment by stevegrossi** [comment]
Yeah, although I think your confusion is understandable. It's definitely weird that you could defeat all of an opponent's units but they're still active. That feels like a point in favor of requiring at least one unit to hold a tile.

**Comment by yakryder** [comment]
_Yeah, although I think your confusion is understandable. It's definitely weird that you could defeat all of an opponent's units but they're still active. That feels like a point in favor of requiring at least one unit to hold a tile._

We could do that. Or we could always make the player defeat condition be no owned territories and no reserve units

**Comment by yakryder** [comment]
actually probably that approach would be no owned territories or no units anywhere (map or reserve)

**Comment by stevegrossi** [comment]
> We could do that. Or we could always make the player defeat condition be no owned territories and no reserve units

Ooh, or what if you defeat a player you get to keep any units they have in reserve? Kind of like how in Risk you get the leftover cards of anyone you defeat.

**Comment by yakryder** [comment]
I like that. I think it might prevent players from dragging things out in an unfun way in some scenarios
